src/Token.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types=1);
+
+namespace MyOnlineStore\GuzzleAuthorizationMiddleware;
+
+final class Token
+{
+    /**
+     * @var string
+     */
+    private $token;
+
+    /**
+     * @var \DateTimeImmutable|null
+     */
+    private $expiresAt;
+
+    public function __construct(string $token, \DateTimeImmutable $expiresAt = null)
+    {
+        $this->token = $token;
+        $this->expiresAt = $expiresAt;
+    }
+
+    public function getToken(): string
+    {
+        return $this->token;
+    }
+
+    /**
+     * @return \DateTimeImmutable|null
+     */
+    public function getExpiresAt()
+    {
+        return $this->expiresAt;
+    }
+
+    public function isExpired(): bool
+    {
+        if (null === $this->expiresAt) {
+            return false;
```

Comment:
Should it be possible to have an infinitely valid token?

Reply:
Depends on the kind of token. I chose this solution to keep things as generic as possible. I could agree on enforcing an expiry time as well though.

Reply:
> I chose this solution to keep things as generic as possible

Perhaps you could introduce a (constructor) parameter on whether to allow tokens without expiry?

Reply:
What would be the added benefit of an additional parameter, instead of just using the nullable expiresAt field?

Reply:
That you keep the "generic" solution: that the package supports having infinitely valid tokens, yet also offers the possibility of blocking them.

What I meant was something like:
```
/**
 * @var bool
 */
private $allowNonExpiringTokens;

//....
if (null === $this->expiresAt) {
      return !$this->allowNonExpiringTokens;
}
```